code/game/objects/items/weapons/chemical_flamethrower/fire_effect.dm

```diff
@@ -0,0 +1,106 @@
+GLOBAL_LIST_EMPTY(flame_effects)
+#define MAX_FIRE_EXIST_TIME 10 MINUTES // That's a lot of fuel, but you are not gonna make it last for longer
+
+/obj/effect/fire
+	name = "fire"
+	desc = "You don't think you should touch this."
+	icon = 'icons/effects/chemical_fire.dmi'
+	icon_state = "fire1"
+
+	/// How hot is our fire?
+	var/temperature
+	/// How long will our fire last
+	var/duration = 10 SECONDS
+	/// How many firestacks does the fire give to mobs
+	var/application_stacks = 1
+
+/obj/effect/fire/Initialize(mapload, reagent_temperature, reagent_duration, fire_applications)
+	. = ..()
+
+	if(!reagent_duration || !reagent_temperature) // There is no reason for this thing to exist
+		qdel(src)
+		return
+
+	duration = reagent_duration
+	temperature = reagent_temperature
+	application_stacks = max(application_stacks, fire_applications)
+
+	for(var/obj/effect/fire/flame as anything in GLOB.flame_effects)
```

Comment:
Why not just check `get_turf(src)` here? Is that any slower than this?

Reply:
No I'm just an idiot

Reply:
Like, unless a fire is put on a turf with a few hundred items, `get_turf(src)` should be a lot faster